test/python/fe_api/grouped_gemm/test_grouped_gemm_quant_jax.py

```
@@ -0,0 +1,87 @@
+import pytest
+
+jax = pytest.importorskip("jax")
+ml_dtypes = pytest.importorskip("ml_dtypes")
+import jax.numpy as jnp
+
+
+def make_jax_inputs(m=256, n=256, k=128, experts=2):
+    """Plausible (physical-layout) inputs; rejection fires before shape validation."""
+    rng = np.random.default_rng(20260809)
+    a_j = jnp.asarray(rng.integers(0, 255, (m, k, 1), dtype=np.uint8).view(ml_dtypes.float8_e4m3fn))
+    rest_k = -(-(-(-k // 32) // 4))  # ceil_div(ceil_div(k, 32), 4)
```

Ruff (0.16.1)

[warning] Python does not support the unary prefix decrement operator (`--`)

(B002)